src/components/ListItem.js

```diff
@@ -0,0 +1,46 @@
+import { doc, updateDoc } from 'firebase/firestore';
+import { db } from '../lib/firebase';
+import { useEffect, useState } from 'react';
+
+const oneDayInSeconds = 86400;
+
+const style = {
+       listStyleType: 'none',
+       textAlign: 'left'
+}
+const ListItem = ({ itemData }) => {
+    const [checked, setChecked] = useState(itemData.lastPurchased !== null)
+
+    const wasPurchasedWithin24Hours = () => {
+        if (itemData.lastPurchased === null) {
+            return false
+        }
+        return (Math.floor(Date.now() / 1000 ) - itemData.lastPurchased.seconds) <= oneDayInSeconds
+    }
+
+    useEffect(() => {
+        if (itemData.lastPurchased === null) {
+            return false
+        }
+           
+        if ((Math.floor(Date.now() / 1000 ) - itemData.lastPurchased.seconds) >= oneDayInSeconds) {
+            setChecked(false)
+        } 
+      }, [itemData]);
+
+    const handleChange = () => {
+        const docRef = doc(db, "Lists", itemData.id)
+        if (itemData.lastPurchased === null || (Math.floor(Date.now() / 1000 ) - itemData.lastPurchased.seconds) >= oneDayInSeconds)  {
```

Comment:
This is a very big if condition, maybe it can be made into a function that also describes the logic of the condition :slightly_smiling_face: ?